### PR TITLE
[TextGeneration] Add helper function to parse model path from args

### PR DESCRIPTION
**Pull request by dsikka**
# Summary
- Add a quick helper function which supports non-kwarg for model_path when using the `TextGeneration` pathway
- For this ticket: https://app.asana.com/0/1206109050183159/1206524727025314/f

The following now works:

```python
from deepsparse import TextGeneration
model_path = "hf:mgoin/TinyStories-1M-ds"

pipe = TextGeneration(model_path)

```

This still works:

```python

model_path  = TextGeneration(model_path=model_path)
```

### CODE REVIEW

**Review by bfineran — DISMISSED** [review]

File: src/deepsparse/pipeline.py [hunk 1]
```diff
@@ -802,3 +805,14 @@ def zero_shot_text_classification_pipeline(*args, **kwargs) -> "Pipeline":
...
+
+
+def _check_model_path_arg(*args, **kwargs):
+    if args:
+        if len(args) > 1 or "model_path" in kwargs or "model" in kwargs:
+            raise ValueError(
+                "Only the model path can be provided as a non-kwarg argument"
+            )
+        kwargs["model_path"] = args[0]
+        args = args[1:]
```
Comment (rahul-tuli) [inline src/deepsparse/pipeline.py:817]: Won't args be an empty list here?

File: src/deepsparse/pipeline.py [hunk 2]
```diff
@@ -802,3 +805,14 @@ def zero_shot_text_classification_pipeline(*args, **kwargs) -> "Pipeline":
     is returned depends on the value of the passed model_scheme argument.
     """
     return Pipeline.create("zero_shot_text_classification", *args, **kwargs)
+
+
+def _check_model_path_arg(*args, **kwargs):
+    if args:
+        if len(args) > 1 or "model_path" in kwargs or "model" in kwargs:
```
Comment (rahul-tuli) [inline src/deepsparse/pipeline.py:812]: why not have the functions accept `model_path` as an arg directly? then python would raise an error automatically if kwargs also have model_path, we'll still have to check for `model` in kwargs though? Thoughts?


for example changing:

```python
def chat_pipeline(*args, **kwargs) -> "Pipeline":
```

To
```python
def chat_pipeline(model_path, **kwargs) -> "Pipeline":
```

Comment (dsikka) [reply src/deepsparse/pipeline.py:812]: Can't just have `model_path` as we also supprt `model` so seems cleaner to just accept args and parse?

File: src/deepsparse/pipeline.py (hunk 1 above)
Comment (dsikka) [reply src/deepsparse/pipeline.py:817]: Yah good catch. Shouldnt be any other args.

**Review by mgoin — APPROVED** [review]

**Pull request merged**